models/articles-models.js

```diff
@@ -60,3 +57,22 @@ exports.selectCommentsByArticleId = (id) => {
       }
     });
 };
+
+exports.insertComment = (id, { author, body }) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [id])
+    .then(({ rows }) => {
+      if (!rows.length) {
+        return Promise.reject({ status: 404, msg: "Article does not exist" });
+      } else {
+        return db
+          .query(
+            `INSERT INTO comments (article_id, author, body) VALUES ($1, $2, $3) RETURNING *`,
+            [id, author, body]
+          )
+          .then(({ rows }) => {
+            return rows[0];
+          });
+      }
```

Comment:
Similar to last ticket. It does work, but we have a nested then block. But you now know how to fix this 😁